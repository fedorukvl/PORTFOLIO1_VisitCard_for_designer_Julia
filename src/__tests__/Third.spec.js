import React from "react";
import { shallow } from "enzyme";
import { Third } from "../Components/index.js";

describe("Tests of <Third/> component", () => {
	let thirdComponent;
	beforeEach(() => {
		thirdComponent = shallow(<Third />);
	});
	afterEach(() => {
		thirdComponent = null;
	});
	describe("first render of <Third/> component", () => {
		it("<Third> component renders properly", () => {
			expect(thirdComponent).not.toBeNull();
			expect(thirdComponent.debug()).toMatchSnapshot();
		});
	});

	describe("<Third/> component renders elements inside itself", () => {
		it("<Third/> component has elements inside", () => {
			expect(thirdComponent.find(".karateDiv").children()).toHaveLength(
				2
			);
			expect(
				thirdComponent.find(".karateDiv").last().find("h1").text()
			).toEqual("Примеры выполненных эскизов на тему : ");
		});
	});

	describe("checking <Third/> component`s props", () => {
		describe("<Third/> wihout props", () => {
			it("without props <Third/> component hasn`t <p> in <div .rightDescriptionDiv>", () => {
				expect(
					thirdComponent.find(".rightDescriptionDiv").find("p")
				).toHaveLength(0);
			});
		});
	});
});
