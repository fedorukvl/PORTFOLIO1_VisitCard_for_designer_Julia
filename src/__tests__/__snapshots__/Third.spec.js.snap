// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests of <Third/> component first render of <Third/> component <Third> component renders properly 1`] = `
"<div className=\\"karateDiv\\">
  <div className=\\"leftImgDiv\\">
    <img src=\\"5.png\\" alt=\\"karate concept #1\\" />
    <img src=\\"6.png\\" alt=\\"karate concept #2\\" />
    <img src=\\"7.png\\" alt=\\"karate concept #3\\" />
  </div>
  <div className=\\"rightDescriptionDiv\\">
    <h1>
      Примеры выполненных эскизов на тему : 
    </h1>
    <br />
  </div>
</div>"
`;
